### PR TITLE
Enumerables

### PR DESCRIPTION
**Pull request by gabrielyea**
# In this PR we:
Create a class MyList that has an instance variable @list.
In MyList implement a method #each that yields successive members of @list and uses the MyEnumerable module.
Create a module MyEnumerable that implements the following methods:
- all?
- any?
- filter
Each class and module has a separate .rb file.

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  def initialize(list)
+    @list = list
+  end
+
+  def each
+    @list.each { |e| yield(e) if block_given? }
+  end
+end
+
+list = MyList.new([1, 2, 3, 4])
```
Comment (vichuge) [inline my_list.rb]: - [x] Please, make changes to use `list = MyList.new(1, 2, 3, 4)` in place an array according the [requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/simple-ruby/create_your_own_enumerable.md#:~:text=irb%3E%20list%20%3D%20MyList.new(1%2C%202%2C%203%2C%204)). 🐱

**Review by vichuge — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ⚠️ 🚦 

Hi dear student, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

mario_ladders

## Good points 👍 
You used proper GitHub Flow ✔️
Your linter checks are all passing ✔️
Your PR messages are descriptive ✔️

## Aspects to improve ♻️

- [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: my_list.rb (hunk 1 above)
Comment (gabrielyea) [reply my_list.rb]: Ups, didn't realize that! 
Its fixed now!

**Review by vichuge — APPROVED** [review]
# Approved project! ✅ 🥳 🎈 🎉🎉🎉🎉🎉🎉🎉🎉 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

InbornCandidHornet-size_restricted

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**